fbw-a380x/src/systems/instruments/src/MFD/pages/common/InputField.tsx

```diff
@@ -0,0 +1,604 @@
+  disabled?: Subscribable<boolean>;
+  canBeCleared?: Subscribable<boolean>;
+  /** Value will be displayed in smaller font, if not entered by pilot (i.e. computed) */
+  enteredByPilot?: Subscribable<boolean>;
+  canOverflow?: boolean;
+  value: Subject<T | null> | Subscribable<T | null>;
+  /** If defined, this component does not update the value prop, but rather calls this method. */
+  onModified?: (newValue: T | null) => void;
+  /** Called for every character that is being typed */
+  onInput?: (newValue: string) => void;
+  /**
+   * Function which modifies data within flight plan. Called during validation phase, after data entry format has been checked
+   * @param newValue to be validated
+   * @returns whether validation was successful. If nothing is returned, success is assumed
+   */
+  dataHandlerDuringValidation?: (newValue: T | null, oldValue?: T | null) => Promise<boolean | void>;
+  errorHandler?: (errorType: FmsErrorType) => void;
+  handleFocusBlurExternally?: boolean;
+  containerStyle?: string;
+  alignText?: 'flex-start' | 'center' | 'flex-end' | Subscribable<'flex-start' | 'center' | 'flex-end'>;
+  tmpyActive?: Subscribable<boolean>;
+  /** Only handles KCCU input for respective side, receives key name only */
+  hEventConsumer: Consumer<string>;
+  /** Kccu uses the HW keys, and doesn't focus input fields */
+  interactionMode: Subscribable<InteractionMode>;
+  // inViewEvent?: Consumer<boolean>; // Consider activating when we have a larger collision mesh for the screens
+}
+
+/**
+ * Input field for text or numbers
+ */
+export class InputField<T> extends DisplayComponent<InputFieldProps<T>> {
+  // Make sure to collect all subscriptions here, otherwise page navigation doesn't work.
+  private subs = [] as Subscription[];
+
+  private readonly guid = `InputField-${Utils.generateGUID()}`;
+
+  public topRef = FSComponent.createRef<HTMLDivElement>();
+
+  public containerRef = FSComponent.createRef<HTMLDivElement>();
+
+  private spanningDivRef = FSComponent.createRef<HTMLDivElement>();
+
+  public textInputRef = FSComponent.createRef<HTMLSpanElement>();
+
+  private caretRef = FSComponent.createRef<HTMLSpanElement>();
+
+  private leadingUnit = Subject.create<string>('');
+
+  private trailingUnit = Subject.create<string>('');
+
+  private leadingUnitRef = FSComponent.createRef<HTMLSpanElement>();
+
+  private trailingUnitRef = FSComponent.createRef<HTMLSpanElement>();
+
+  private modifiedFieldValue = Subject.create<string | null>(null);
+
+  private isFocused = Subject.create(false);
+
+  private isValidating = Subject.create(false);
+
+  private alignTextSub: Subscribable<'flex-start' | 'center' | 'flex-end'> = SubscribableUtils.toSubscribable(
+    this.props.alignText ?? 'center',
+    true,
+  );
+
+  private onNewValue() {
+    // Don't update if field is being edited
+    if (this.isFocused.get() === true || this.isValidating.get() === true) {
```

Comment:
There is a bunch of these.

```suggestion
    if (this.isFocused.get() || this.isValidating.get()) {
```